pkg/operator/ceph/cluster/controller.go

```diff
@@ -218,15 +222,17 @@ func add(mgr manager.Manager, r reconcile.Reconciler, context *clusterd.Context)
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileCephCluster) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
 	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
-	reconcileResponse, err := r.reconcile(request)
+	reconcileResponse, cephCluster, err := r.reconcile(request)
+
 	if err != nil {
 		logger.Errorf("failed to reconcile. %v", err)
+		r.clusterController.recorder.Report(cephCluster, cephCluster.Name, "Warning", "FailedReconcile", err.Error())
 	}
 
 	return reconcileResponse, err
 }
 
-func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.Result, *cephv1.CephCluster, error) {
```

Comment:
If we can report the events at the same time as the conditions are updated, it will already have access to the cephCluster, so there would be no need to pass it around in these other locations.

Reply:
I see that we are updating the conditions at many places If we will go that way then we will again break this as we will have to call reorder at many places https://github.com/rook/rook/pull/7222#discussion_r576124033